### PR TITLE
Issue #698 - RsFlash: Get rid of the SingularField PMD suppression

### PR DESCRIPTION
**Pull request by dalifreire**
Issue #698. After upgrade to qulice 0.16.4 we have to fix SingularField PMD warings in the RSFlash class. 
This PR do the following changes:
- Removes the SingularField PMD supression on RsFlash.java.
- Removes the lombok.ToString and override toString using the text variable

### CODE REVIEW

**Comment by dalifreire** [comment]
@davvd Please, find someone to review.

**Comment by davvd** [comment]
@dalifreire I will find a reviewer for your pull requests shortly, thanks for contribution!

**Comment by davvd** [comment]
@mkordas please review, thanks

**Comment by mkordas** [comment]
@dalifreire I'll check it soon

**Comment by mkordas** [comment]
@dalifreire I'm on it :)

File: src/main/java/org/takes/facets/flash/RsFlash.java [hunk 1]
```diff
@@ -100,12 +99,7 @@
  * @author Yegor Bugayenko (yegor@teamed.io)
  * @version $Id$
  * @since 0.1
- * @todo #686:15min After upgrade to qulice 0.16.4 we have to fix
- *  SingularField PMD warings in this class. Fix this warning
- *  and rid of the PMD suppression here.
  */
-@SuppressWarnings("PMD.SingularField")
-@ToString(callSuper = true, of = "text")
```
Comment (mkordas) [inline src/main/java/org/takes/facets/flash/RsFlash.java]: @dalifreire how `callSuper = true` is replaced in the current implementation?

File: src/main/java/org/takes/facets/flash/RsFlash.java [hunk 2]
```diff
@@ -195,6 +188,11 @@ public RsFlash(final CharSequence msg, final Level level,
         this.text = String.format(RsFlash.TEXT_FORMAT, level, msg);
     }
 
+    @Override
+    public String toString() {
+        return String.valueOf(this.text);
```
Comment (mkordas) [inline src/main/java/org/takes/facets/flash/RsFlash.java]: @dalifreire is this enough? What e.g about class name?

File: src/main/java/org/takes/facets/flash/RsFlash.java [hunk 3]
```diff
@@ -117,7 +111,6 @@
     /**
      * To string.
      */
-    @SuppressWarnings("unused")
     private final transient CharSequence text;
```
Comment (mkordas) [inline src/main/java/org/takes/facets/flash/RsFlash.java]: @dalifreire so the only purpose of this field is to be in `toString`?

**Comment by mkordas** [comment]
@dalifreire see my comments above

File: src/main/java/org/takes/facets/flash/RsFlash.java (hunk 1 above)
Comment (dalifreire) [inline src/main/java/org/takes/facets/flash/RsFlash.java]: @mkordas I think that's not necessary once the superclasses don't have non-static fields.

File: src/main/java/org/takes/facets/flash/RsFlash.java (hunk 2 above)
Comment (dalifreire) [inline src/main/java/org/takes/facets/flash/RsFlash.java]: @mkordas I added the class name now.

File: src/main/java/org/takes/facets/flash/RsFlash.java (hunk 3 above)
Comment (dalifreire) [inline src/main/java/org/takes/facets/flash/RsFlash.java]: @mkordas Yes.

**Comment by dalifreire** [comment]
@mkordas Thanks for your review. Please, see my comments above. What do you think now?

**Comment by mkordas** [comment]
@dalifreire checking :)

File: src/main/java/org/takes/facets/flash/RsFlash.java (hunk 1 above)
Comment (mkordas) [inline src/main/java/org/takes/facets/flash/RsFlash.java]: @dalifreire what about https://github.com/yegor256/takes/blob/master/src/main/java/org/takes/rs/RsWrap.java#L48?

File: src/main/java/org/takes/facets/flash/RsFlash.java (hunk 3 above)
Comment (mkordas) [inline src/main/java/org/takes/facets/flash/RsFlash.java]: @dalifreire so where this `toString` is used?

File: src/main/java/org/takes/facets/flash/RsFlash.java (hunk 2 above)
Comment (mkordas) [inline src/main/java/org/takes/facets/flash/RsFlash.java]: @dalifreire does it follow the format used by Lombok?

**Comment by mkordas** [comment]
@dalifreire responded :)

File: src/main/java/org/takes/facets/flash/RsFlash.java (hunk 1 above)
Comment (dalifreire) [inline src/main/java/org/takes/facets/flash/RsFlash.java]: @mkordas you are right about non-static field, thanks. I have changed the toString method.

File: src/main/java/org/takes/facets/flash/RsFlash.java (hunk 2 above)
Comment (dalifreire) [inline src/main/java/org/takes/facets/flash/RsFlash.java]: @mkordas Yes, now we have exactly the same format used by Lombok.

File: src/main/java/org/takes/facets/flash/RsFlash.java (hunk 3 above)
Comment (dalifreire) [inline src/main/java/org/takes/facets/flash/RsFlash.java]: @mkordas It's a public method, but I did not find any place where this toString method is used (this method was already there (lombok.ToString annotation)).

**Comment by dalifreire** [comment]
@mkordas thanks again!
So, I don't understand which is the point that you are going... the issue #698 it's about 15 minutes to remove the SingularField PMD supression.. did you read the issue #698?
I thought that the better option to get rid of the SingularFieldd PMD suppression is remove the lombok.ToString annotation... do you suggest a different approach?

**Comment by mkordas** [comment]
@dalifreire on it again

**Comment by mkordas** [comment]
@dalifreire 
> I thought that the better option to get rid of the SingularFieldd PMD suppression is remove the lombok.ToString annotation... do you suggest a different approach?

No, approach is fine. I just struggle to understand how this class is used and what are consequences of changing `toString()` as anyone could be relying on it (since it is public)

File: src/main/java/org/takes/facets/flash/RsFlash.java (hunk 1 above)
Comment (mkordas) [inline src/main/java/org/takes/facets/flash/RsFlash.java]: @dalifreire looks good, thanks

File: src/main/java/org/takes/facets/flash/RsFlash.java (hunk 3 above)
Comment (mkordas) [inline src/main/java/org/takes/facets/flash/RsFlash.java]: @dalifreire strange, so what does this class do? Nothing? Maybe their only responsibility is to provide constructors and actual behaviour is defined in parents

File: src/main/java/org/takes/facets/flash/RsFlash.java [hunk 4]
```diff
@@ -195,6 +188,16 @@ public RsFlash(final CharSequence msg, final Level level,
         this.text = String.format(RsFlash.TEXT_FORMAT, level, msg);
     }
 
+    @Override
+    public String toString() {
+        return String.format(
+            "%s(super=%s, text=%s)",
+            RsFlash.class.getSimpleName(),
+            super.toString(),
+            this.text
```
Comment (mkordas) [inline src/main/java/org/takes/facets/flash/RsFlash.java]: @dalifreire can we join these lines as much as possible?

**Comment by mkordas** [comment]
@dalifreire see my responses above

File: src/main/java/org/takes/facets/flash/RsFlash.java (hunk 3 above)
Comment (dalifreire) [inline src/main/java/org/takes/facets/flash/RsFlash.java]: @mkordas Yes, I think the same as you. This class (and the text field too) was created by @yegor256.

File: src/main/java/org/takes/facets/flash/RsFlash.java (hunk 4 above)
Comment (dalifreire) [inline src/main/java/org/takes/facets/flash/RsFlash.java]: @mkordas Done!

**Comment by dalifreire** [comment]
@mkordas Please, could you take a look if everything is right now?

**Comment by mkordas** [comment]
@dalifreire yes, looks good. Having a test for this method would also be nice addition, but I leave this decision to @yegor256

**Comment by mkordas** [comment]
@rultor merge

**Comment by dalifreire** [comment]
@mkordas Thanks for your review again.

**Comment by mkordas** [comment]
@yegor256 we need your review here and decision about test

**Comment by mkordas** [comment]
@yegor256 ping

**Comment by mkordas** [comment]
@yegor256 ping again

**Comment by mkordas** [comment]
@yegor256 please review

**Comment by yegor256** [comment]
@rultor merge

**Comment by davvd** [comment]
@elenavolokhova please, review this issue for QA compliance, as per [par.24](http://at.teamed.io/policy.html#24)

**Comment by davvd** [comment]
@rultor deploy now pls

**Comment by elenavolokhova** [comment]
@davvd Looks good!

**Comment by davvd** [comment]
> @davvd Looks good!

@elenavolokhova thanks, got it, great to see high-quality tickets here

**Comment by davvd** [comment]
@mkordas 10 mins was added to the account of @elenavolokhova (for QA review), in transaction 97247087

paid, thanks, added **38 mins** to your account, payment `57b597884e60760910997606`, 202 hours and 18 mins was spent

review comments (c=23) added as a bonus

+38 added to your rating, at the moment it is: [+5120](http://www.netbout.com/b/36080?open=rating)

**Pull request merged**